norns/dev-setup.md

```diff
@@ -0,0 +1,146 @@
+---
+layout: default
+parent: scripting
+grand_parent: norns
+has_children: false
+title: dev setup
+nav_order: 3
+has_toc: false
+---
+
+# norns dev setup
+
+if using [maiden](../maiden) is plenty sufficient to create norns script, some experienced users may want to rely on development tools they are already familiar with.
+
+
+## remote access via command line
+
+### basics
+
+when accessing norns via the command line (using [SSH / USB serial access](../advanced-access)), scripts can directly be edited this way using locally-installed editors (such an `nano` or `vi`).
+
+this also allows interacting with [maiden from the command line](../maiden/#advanced-access)
+
+this video is a walkthrough of this process:
+
+<iframe width="560" height="315" src="https://www.youtube.com/embed/S-jhIT4mCcM" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+
+
+### sshfs - mount source files locally
+
+`sshfs` allow to locally mount norns filesystem onto a development machine as if it were local. this allows you to use any prefered editor / tools installed locally.
```

Comment:
fwiw using `sshfs` has grown more difficult on macos machines due to changes to macfuse becoming closed source. these instructions seem directed towards editing files in the dust directory which is already exposed via `smb`. i would recommend we direct people that direction instead of `sshfs`

Reply:
IIRC it was Tyler that had recommended this approach at the time.

@tyleretters, was `sshfs` providing any advantage over accessing files via a `samba` mountpoint?

Reply:
i found it to be easier to work with. i've been able to get sshfs working on macos, but agree with @ngwese that it isn't something i'd direct towards even intermediate folks.

Reply:
:) i have a vague recollection i have been the one who pointed @tyleretters at `sshfs` back in a different channel when the topic of how people go about doing dev directly on a norns device came up.

it's not that `sshfs` is bad (i personally prefer it over using `smb`) yet if the docs point to is as a solution it is valuable to provide some guidance as to when/why someone might choose to use it over the existing `smb` support. i use `sshfs` to gain access to any part of the norns filesystem i need, unless one is hacking on the core component it isn't something which is really needed.

Reply:
ah, i see, so the samba mountpoint is restricted to `~/dust` which is plenty sufficient for casual norns scripting.
but accessing norns core is quite useful when doing stuff like mod development.

now that `maiden-run` got removed, maybe should this section be renamed to something like "mods and norns core development"?

Reply:
ah yeah, in this context, if we ever promoted `sshfs` it should go under this same `./norns/advanced-access` page, not the `dev-setup` one proposed in this PR.

so i proposed we nuke the `remote access via command line` section from this proposed page.

Reply:
nuked it